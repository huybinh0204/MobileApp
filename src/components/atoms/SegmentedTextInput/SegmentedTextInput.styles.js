import styled from '@emotion/native';
import BaseText from '_components/atoms/BaseText/BaseText';
import { normalize } from '_utilities/screen';

export const Cell = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.beta900,
  lineHeight: theme.spacing.l,
  fontSize: theme.fontSize.title,
  paddingTop: theme.spacing.s,
  textAlign: 'center',
  width: theme.spacing.l,
  height: normalize(75),
}));

export const CellContainer = styled.View(({ theme }) => ({
  borderColor: theme.colors.beta100,
  borderRadius: 6,
  borderWidth: 2,
  zIndex: 0,
}));

export const Placeholder = styled(BaseText)(({ theme }) => ({
  color: theme.colors.beta200,
}));

export const Separator = styled.View(({ theme }) => ({
  alignSelf: 'center',
  backgroundColor: theme.colors.beta200,
  height: 2,
  width: 6,
}));
