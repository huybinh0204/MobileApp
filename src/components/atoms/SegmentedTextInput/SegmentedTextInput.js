import PropTypes from 'prop-types';
import React from 'react';
import {
  CodeField,
  useBlurOnFulfill,
  useClearByFocusCell,
} from 'react-native-confirmation-code-field';
import SegmentedTextInputCell from './SegmentedTextInputCell';

const SegmentedTextInput = ({
  cellCount,
  placeholder,
  separatorIndexes,
  setValue,
  value,
  ...rest
}) => {
  const ref = useBlurOnFulfill({ value, cellCount });
  const [props, getCellOnLayoutHandler] = useClearByFocusCell({ value, setValue });

  return (
    <CodeField
      {...rest}
      {...props}
      ref={ref}
      cellCount={cellCount}
      onChangeText={setValue}
      value={value}
      renderCell={({ index, symbol, isFocused }) => (
        <SegmentedTextInputCell
          isFocused={isFocused}
          index={index}
          key={index}
          onLayout={getCellOnLayoutHandler(index)}
          placeholder={placeholder}
          separatorIndexes={separatorIndexes}
          symbol={symbol}
        />
      )}
    />
  );
};

SegmentedTextInput.defaultProps = {
  placeholder: null,
  separatorIndexes: [],
};

SegmentedTextInput.propTypes = {
  cellCount: PropTypes.number.isRequired,
  placeholder: PropTypes.string,
  separatorIndexes: PropTypes.arrayOf(PropTypes.number),
  setValue: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default SegmentedTextInput;
