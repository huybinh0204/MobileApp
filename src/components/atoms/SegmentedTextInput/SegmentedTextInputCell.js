import PropTypes from 'prop-types';
import React from 'react';
import { Cursor } from 'react-native-confirmation-code-field';
import { Cell, CellContainer, Placeholder, Separator } from './SegmentedTextInput.styles';

const CellContent = ({ isFocused, index, placeholder, symbol }) => {
  if (symbol) return symbol;

  if (isFocused) return <Cursor />;

  return placeholder ? <Placeholder>{placeholder[index]}</Placeholder> : null;
};

const SegmentedTextInputCell = ({
  isFocused,
  index,
  onLayout,
  placeholder,
  separatorIndexes,
  symbol,
}) => (
  <>
    <CellContainer key={`value-${index}`} onLayout={onLayout}>
      <Cell>
        <CellContent
          isFocused={isFocused}
          index={index}
          placeholder={placeholder}
          symbol={symbol}
        />
      </Cell>
    </CellContainer>
    {separatorIndexes.includes(index) ? <Separator key={`separator-${index}`} /> : null}
  </>
);

SegmentedTextInputCell.propTypes = {
  placeholder: null,
  separatorIndexes: [],
};

SegmentedTextInputCell.propTypes = {
  isFocused: PropTypes.bool.isRequired,
  index: PropTypes.number.isRequired,
  onLayout: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  separatorIndexes: PropTypes.arrayOf(PropTypes.number),
  symbol: PropTypes.string.isRequired,
};

export default SegmentedTextInputCell;
