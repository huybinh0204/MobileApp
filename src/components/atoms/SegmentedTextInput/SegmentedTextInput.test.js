import { render } from '@testing-library/react-native';
import React from 'react';
import Theme from '_utilities/Theme';
import SegmentedTextInput from './SegmentedTextInput';

describe('SegmentedTextInput', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <Theme>
        <SegmentedTextInput
          cellCount={10}
          placeholder="XXXXXXXXXX"
          separatorIndexes={[]}
          setValue={() => {}}
          value="value"
        />
      </Theme>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
