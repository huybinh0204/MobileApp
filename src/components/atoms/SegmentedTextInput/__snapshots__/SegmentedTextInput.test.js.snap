// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedTextInput should match the snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      v
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      a
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      l
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      u
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      e
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#BFC4C8",
            },
            undefined,
          ]
        }
      >
        X
      </Text>
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#BFC4C8",
            },
            undefined,
          ]
        }
      >
        X
      </Text>
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#BFC4C8",
            },
            undefined,
          ]
        }
      >
        X
      </Text>
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#BFC4C8",
            },
            undefined,
          ]
        }
      >
        X
      </Text>
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D4D7DB",
          "borderRadius": 6,
          "borderWidth": 2,
          "zIndex": 0,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "fontFamily": "Poppins",
            "fontSize": 36,
            "height": 136,
            "lineHeight": 87,
            "paddingTop": 29,
            "textAlign": "center",
            "width": 87,
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#BFC4C8",
            },
            undefined,
          ]
        }
      >
        X
      </Text>
    </Text>
  </View>
  <TextInput
    autoCapitalize="characters"
    autoCorrect={false}
    blurOnSubmit={false}
    caretHidden={true}
    clearButtonMode="never"
    disableFullscreenUI={true}
    maxLength={10}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onPressOut={[Function]}
    spellCheck={false}
    style={
      Object {
        "bottom": 0,
        "fontSize": 1,
        "left": 0,
        "opacity": 0.01,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    underlineColorAndroid="transparent"
    value="value"
  />
</View>
`;
